sqlmesh/core/dialect.py

```diff
@@ -294,15 +294,17 @@ def _parse_props(self: Parser) -> t.Optional[exp.Expression]:
     if not key:
         return None
 
-    if self._match(TokenType.L_PAREN):
-        value: t.Optional[exp.Expression] = self.expression(
+    name = key.name.lower()
+    if name == "when_matched":
+        value: t.Optional[exp.Expression] = self._parse_when_matched()[0]
```

Comment:
can't there be multiple when matched expressions? 

when matched, when not matched etc

Reply:
This PR is just implementing when matched. I believe that is all that is needed for now and we can add more later if that is required.

Reply:
you don't think it's worth it to just implement both? complexity doesn't seem too high